inc/toolbox.class.php

```diff
@@ -3490,4 +3492,30 @@ public static function doubleEncodeEmails($string) {
       }, $string);
       return $string;
    }
+
+   /**
+    * Normalizes file name
+    *
+    * @param string filename
+    *
+    * @return string
+    */
+   public static function filename($filename): string {
+      //remove extension
+      $ext = pathinfo($filename, PATHINFO_EXTENSION);
+      $filename = self::slugify(
+         preg_replace(
+            '/\.' . $ext . '$/',
+            '',
+            $filename
+         ),
+         '' //no prefix on filenames
+      );
+
+      if (!empty($ext)) {
+         $filename .= '.' . $ext;
+      }
+
+      return $filename;
```

Comment:
According to https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits , it could be a good idea to limit filename to 255 chars.